src/models/addressBook/index.js

```diff
@@ -0,0 +1,132 @@
+import { joinPath } from "../../stringHelpers";
+import { chain } from "../../solidClientHelpers/utils";
+import { ERROR_CODES, isHTTPError } from "../../error";
+import { vcardExtras } from "../../addressBook";
+
+/**
+ * @typedef AddressBook
+ * @type {object}
+ * @property {string} containerIri - The address to the container
+ * @property {object} dataset - The dataset the main index is stored in
+ * @property {object} thing - The thing that represents the main index itself
+ */
+
+/* Model constants */
+const CONTACTS_CONTAINER = "contacts/";
+
+export const ADDRESS_BOOK_ERROR_ALREADY_EXIST = "Address book already exists.";
+export const ADDRESS_BOOK_ERROR_NO_MAIN_INDEX = "Unable to load main index";
+export const ADDRESS_BOOK_ERROR_NO_PERMISSION_TO_CREATE =
+  "You do not have permission to create an address book";
+
+/* Model functions */
+export function getAddressBookContainerIri(podRootIri) {
+  return joinPath(podRootIri, CONTACTS_CONTAINER);
+}
+
+export function getAddressBookIndexDefaultUrl(containerIri) {
+  return joinPath(containerIri, "index.ttl");
+}
+
+export function getAddressBookIndexUrl(addressBook) {
+  return (
+    getSourceUrl(addressBook.dataset) ||
+    getAddressBookIndexDefaultUrl(addressBook.containerIri)
+  );
+}
+
+export function createAddressBook(containerIri, owner, title = "Contacts") {
+  const thing = chain(
+    createThing({ name: "this" }),
+    (t) => addUrl(t, rdf.type, vcardExtras("AddressBook")),
+    (t) => addUrl(t, acl.owner, owner),
+    (t) => addStringNoLocale(t, dc.title, title)
+  );
+  return {
+    containerIri,
+    dataset: setThing(createSolidDataset(), thing),
+    thing,
+  };
+}
+
+export async function loadAddressBook(containerIri, fetch) {
+  const mainIndexUrl = getAddressBookIndexDefaultUrl(containerIri);
+  const mainIndexDataset = await getSolidDataset(mainIndexUrl, { fetch });
+  const mainIndexThingUrl = `${mainIndexUrl}#this`;
+  const mainIndex = getThing(mainIndexDataset, mainIndexThingUrl);
+  if (!mainIndex) {
+    throw new Error(ADDRESS_BOOK_ERROR_NO_MAIN_INDEX);
+  }
+  return {
+    containerIri,
+    dataset: mainIndexDataset,
+    thing: mainIndex,
+  };
+}
+
+export async function saveNewAddressBook(
+  containerIri,
+  owner,
+  fetch,
+  title = "Contacts"
+) {
+  try {
+    await getSolidDataset(containerIri, { fetch });
+    throw new Error(ADDRESS_BOOK_ERROR_ALREADY_EXIST);
+  } catch (error) {
+    if (!isHTTPError(error, ERROR_CODES.NOT_FOUND)) throw error;
+  }
+
+  const newAddressBook = createAddressBook(containerIri, owner, title);
+  const mainIndexUrl = getAddressBookIndexUrl(newAddressBook);
+
+  try {
+    const dataset = await saveSolidDatasetAt(
+      mainIndexUrl,
+      newAddressBook.dataset,
+      {
+        fetch,
+      }
+    );
+    const thing = getThing(dataset, `${getSourceUrl(dataset)}#this`);
+    return { containerIri, dataset, thing };
+  } catch (error) {
+    if (isHTTPError(error, ERROR_CODES.UNAUTHORIZED)) {
```

Comment:
Since this try/catch only cares whether it's HTTP or it re-throws, you should be able to only wrap the `saveSolidDatasetAt` call.